ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,137 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Example of setInterval
+var init_interval
+
+// Record timer waiting to run
+var exist_timer = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const removeTimer = (key, DB) => {
+    try {
+        //Delete thread when thread exists
+        if (exist_timer[key] != undefined ) {
+            clearTimeout(exist_timer[key].timer)
+        }
+        delete exist_timer[key];
+        // An insurance mechanism to ensure that thread init_interval is active
+        init_secret_delete_thread(DB)
+    } catch (error) {
+        console.error('error removeTimer :: ', error)
+    }
+}
+
+/**
+ * Expose the interface externally, scheduled tasks of delete secret
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const startDeleteSecretTask = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
+        let delayMillisecond = plannedDeleteTime - new Date().getTime()
```

Comment:
Is it possible that delayMillisecond < 0?
For the case that server is down for several minutes or hours and then gets recovered?

Reply:
if the value is less than 0, it will be directly deleted during the first run after the server is started

Reply:
where the code?

Reply:
@setTimeout() if the parameter is negative, it will run directly

Reply:
Thanks.  It depends on the API SetTimeout(). 
But when I look up the API from here: https://nodejs.org/api/timers.html#settimeoutcallback-delay-args
I found a new issue: 
_**When delay is larger than 2147483647 or less than 1, the delay will be set to 1. Non-integer delays are truncated to an integer.**_

2147483647 / 1000 / 60 / 60 / 24 = 24.85 day
But what is the maximum we would support?

Reply:
The maximum is 365 days

Reply:
Thanks     That's fine.